File: src/main/java/kr/hyfata/zero/modules/mailbox/listener/MailboxEventListener.java
package kr.hyfata.zero.modules.mailbox.listener;

import kr.hyfata.zero.modules.mailbox.ZeroMailbox;
import kr.hyfata.zero.modules.mailbox.handler.MailboxHandler;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;

import java.util.concurrent.CompletableFuture;

public class MailboxEventListener implements Listener {
    ZeroMailbox zeroMailbox;
    public MailboxEventListener(ZeroMailbox zeroMailbox) {
        this.zeroMailbox = zeroMailbox;
    }

    @EventHandler
    public void onJoin(PlayerJoinEvent event) {
        CompletableFuture.runAsync(() -> MailboxHandler.sendRemainingMailCount(event.getPlayer()));
    }
}

File: src/main/java/kr/hyfata/zero/modules/mailbox/listener/MailboxEventListener.kt
package kr.hyfata.zero.modules.mailbox.listener

import kr.hyfata.zero.modules.mailbox.ZeroMailbox
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent
import java.util.concurrent.CompletableFuture

class MailboxEventListener(var zeroMailbox: ZeroMailbox) : Listener {
    @EventHandler
    fun onJoin(event: PlayerJoinEvent) {
        CompletableFuture.runAsync { zeroMailbox.handler.sendRemainingMailCount(event.getPlayer()) }
    }
}
